### PR TITLE
Show correct UI when replying to a voice message

### PR DESCRIPTION
**Pull request by julioromano**
Shows voice messages in the room summary.
Shows voice messages in the reply context menu and composer.
Show replies to voice messages in the timeline. 

(before this PR voice messages were shown the same as audio messages)

Story: https://github.com/vector-im/element-meta/issues/2106

### CODE REVIEW

File: libraries/eventformatter/impl/src/test/kotlin/io/element/android/libraries/eventformatter/impl/DefaultRoomLastMessageFormatterTest.kt [hunk 1]
```diff
@@ -228,7 +239,12 @@ class DefaultRoomLastMessageFormatterTest {
             }
             val shouldCreateAnnotatedString = when (type) {
                 is VideoMessageType -> true
-                is AudioMessageType -> true
+                is AudioMessageType -> {
+                    when (type.isVoiceMessage) {
+                        true -> true
+                        false -> true
```
Comment (bmarty) [inline libraries/eventformatter/impl/src/test/kotlin/io/element/android/libraries/eventformatter/impl/DefaultRoomLastMessageFormatterTest.kt:245]: Always `true` then?

**Review by bmarty — COMMENTED** [review]
Thanks, LGTM, will approve after the screenshots have been recorded.

**Comment by julioromano** [comment]
Screenshots: https://github.com/vector-im/element-x-android/actions/runs/6664550538

File: libraries/eventformatter/impl/src/test/kotlin/io/element/android/libraries/eventformatter/impl/DefaultRoomLastMessageFormatterTest.kt (hunk 1 above)
Comment (julioromano) [reply libraries/eventformatter/impl/src/test/kotlin/io/element/android/libraries/eventformatter/impl/DefaultRoomLastMessageFormatterTest.kt:245]: In the future we'll have to split `AudioMessageType` in two (`AudioMessageType` and `VoiceMessageType`).
When we do this having such structure in place might help a bit.

What do you think?
Should I actually split the message type right now?

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/actionlist/ActionListStateProvider.kt [hunk 2]
```diff
@@ -61,7 +63,23 @@ open class ActionListStateProvider : PreviewParameterProvider<ActionListState> {
...
+                    target = ActionListState.Target.Success(
+                        event = aTimelineItemEvent(content = aTimelineItemAudioContent()).copy(
+                            reactionsState = reactionsState
+                        ),
+                        actions = aTimelineItemActionList(),
+                    )
+                ),
+                anActionListState().copy(
+                    target = ActionListState.Target.Success(
+                        event = aTimelineItemEvent(content = aTimelineItemVoiceContent()).copy(
```
[the remark this answers is not in the document]
Comment (julioromano) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/actionlist/ActionListStateProvider.kt:81]: whooops :) fixed

[the remark this answers is not in the document]
Comment (julioromano) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/actionlist/ActionListStateProvider.kt:82]: Though I was also suprised there wasn't a `aTimelineItemAudioContent`

**Review by jmartinesp — APPROVED** [review]
Since @bmarty already gave his approval, I'll approve it now we have some screenshots. There are some missing for the room list and timeline, but we can add them in a separate PR.

**Comment by julioromano** [comment]
> Since @bmarty already gave his approval, I'll approve it now we have some screenshots. There are some missing for the room list and timeline, but we can add them in a separate PR.

Continuation: https://github.com/vector-im/element-x-android/pull/1660

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/actionlist/ActionListStateProvider.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (bmarty) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/actionlist/ActionListStateProvider.kt:82]: Thanks for adding them all!

File: libraries/eventformatter/impl/src/test/kotlin/io/element/android/libraries/eventformatter/impl/DefaultRoomLastMessageFormatterTest.kt (hunk 1 above)
Comment (bmarty) [reply libraries/eventformatter/impl/src/test/kotlin/io/element/android/libraries/eventformatter/impl/DefaultRoomLastMessageFormatterTest.kt:245]: No worries, if there are plan to change this later :)

**Pull request merged**